packages/shared/src/components/post/write/MultipleSourceSelect.tsx

```diff
@@ -0,0 +1,294 @@
+    target="_blank"
+  >
+    <LazyImage
+      imgSrc={defaultSquad.image}
+      imgAlt="New squad image"
+      className="size-8 rounded-full"
+      aria-hidden
+      role="presentation"
+    />
+    <Typography
+      bold
+      className="flex-1"
+      color={TypographyColor.Primary}
+      type={TypographyType.Callout}
+    >
+      Create a new squad
+    </Typography>
+    <ArrowIcon aria-hidden className="rotate-90" />
+  </a>
+);
+
+const MAX_SQUADS_COUNT = 3;
+
+const SourceCheckbox = ({
+  source,
+  disabled = false,
+  ...props
+}: {
+  source: Omit<Squad, 'type'> & { type: SourceType };
+} & CheckboxProps) => {
+  const isUserSource = source.type === SourceType.User;
+  return (
+    <ConditionalWrapper
+      condition={disabled}
+      wrapper={(content) => (
+        <Tooltip
+          content={`You can't choose more than ${MAX_SQUADS_COUNT} squads.`}
+        >
+          <div>{content}</div>
+        </Tooltip>
+      )}
+    >
+      <Checkbox
+        {...props}
+        disabled={disabled}
+        className="min-w-full flex-row-reverse gap-2 !p-0"
+        checkmarkClassName="!mr-0"
+      >
+        <div className="flex items-center">
+          <SourceAvatar source={source} size={ProfileImageSize.Medium} />
+          <div className="flex flex-col">
+            <Typography
+              bold
+              color={
+                disabled ? TypographyColor.Tertiary : TypographyColor.Primary
+              }
+              type={TypographyType.Callout}
+            >
+              {isUserSource ? 'Everyone' : source.name}
+            </Typography>
+            {!isUserSource && source.handle && (
+              <Typography
+                type={TypographyType.Footnote}
+                color={TypographyColor.Tertiary}
+              >
+                @{source.handle}
+              </Typography>
+            )}
+          </div>
+        </div>
+      </Checkbox>
+    </ConditionalWrapper>
+  );
+};
+
+interface SourceMultipleSelectProps {
+  className?: string;
+  selected: string[];
+  setSelected: (selected: string[]) => void;
+}
+
+export const MultipleSourceSelect = ({
+  className,
+  selected,
+  setSelected,
+}: SourceMultipleSelectProps) => {
+  const { user, squads: allSquads } = useAuthContext();
+  const userSource = useMemo(() => generateUserSourceAsSquad(user), [user]);
+  const squads = useMemo(
+    () =>
+      allSquads.filter(
+        (squad) =>
+          squad?.active && verifyPermission(squad, SourcePermissions.Post),
+      ),
+    [allSquads],
+  );
+  const squadsMapById = useMemo(
+    () =>
+      squads.reduce<Record<string, Squad>>((acc, squad) => {
+        acc[squad.id] = squad;
+        return acc;
+      }, {}),
+    [squads],
+  );
+  const selectedSquads = useMemo(
+    () =>
+      selected
+        .filter((sourceId) => sourceId !== userSource.id)
+        .map((sourceId) => squadsMapById[sourceId]),
+    [selected, squadsMapById, userSource.id],
+  );
```

Comment:
- squads: all selectable squads
- squadsMapById: easy way to access squads by ID (in order to get handle, etc...)
- selectedSquads: all the selected elements but the user source